client.go

```diff
@@ -116,6 +124,22 @@ func ClientOptionsFromConf(conf hadoopconf.HadoopConf) ClientOptions {
 		options.KerberosServicePrincipleName = strings.Split(conf["dfs.namenode.kerberos.principal"], "@")[0]
 	}
 
+	dataTransferProt := strings.ToLower(conf["dfs.data.transfer.protection"])
+	switch dataTransferProt {
+	case "privacy":
+		options.EncryptedDataNode = true
+		fallthrough
+	case "integrity", "authentication":
+		options.SecureDataNode = true
+	}
+
+	// dfs.encrypt.data.transfer set to true overrides dfs.data.transfer.protection and
+	// requires both privacy and integrity for communication
+	if conf["dfs.encrypt.data.transfer"] == "true" {
+		options.EncryptedDataNode = true
+		options.SecureDataNode = true
+	}
+
```

Comment:
The comment for this method needs to be updated (to say that we munge these fields)

Reply:
comment updated. let me know if this is good.